includes/Admin/Cross_Origin_Isolation.php

```diff
@@ -81,6 +83,38 @@ public static function get_registration_action_priority(): int {
 		return 11;
 	}
 
+	/**
+	 * Check whether the conditional object is currently needed.
+	 *
+	 * @since 1.6.0
+	 *
+	 * @return bool Whether the conditional object is needed.
+	 */
+	public static function is_needed(): bool {
+		$user_id = get_current_user_id();
+		if ( ! $user_id ) {
+			return false;
+		}
+
+		// Cross-origin isolation is not needed if users can't upload files anyway.
+		if ( ! user_can( $user_id, 'upload_files' ) ) {
+			return false;
+		}
+
+		return rest_sanitize_boolean( Services::get( 'user_preferences' )->get_preference( $user_id, Preferences::MEDIA_OPTIMIZATION_META_KEY ) );
```

Comment:
Same point about dependency injection here. Why not pass an instanse into the class.

Reply:
This line feel a little overly long. How about something like this. 
```suggestion
        $user_preferences = Services::get( 'user_preferences' );
        
		return rest_sanitize_boolean( $user_preferences->get_preference( $user_id, $user_preferences::MEDIA_OPTIMIZATION_META_KEY ) );
```

Reply:
`is_needed()` is a **static** method. Can't use injection there.

Will split the function call across multiple lines.